pkgs/test_core/pubspec.yaml

```diff
@@ -43,3 +43,8 @@ dev_dependencies:
 dependency_overrides:
   test_api:
     path: ./../test_api
+
+  # to remove before publish
+  analyzer: ^0.35.0
+  test:
```

Comment:
There are no tests in this package, and no package:test dependency... :P

Reply:
I know...and yet if you don't add this, you won't get the latest version of pkg:test and few other things...it's weird

Reply:
Ah it picks up a transitive dep on package:test.

The package:test version solve does get the latest analyzer though, and that is where the tests run, so I don't think this is necessary.

Reply:
we'll nix all of the this before publish anyway 🤷‍♂️

Reply:
Lol so this package has a bunch of dev deps, but no tests. As far as I can tell they can all be deleted which removes the circular dep.